### PR TITLE
Add celo to all safes spell

### PR DESCRIPTION
**Pull request by danielpartida**
# SPELLBOOK FREEZE

From June 22 to June 29, we will be freezing some Spellbook contributions for the migration to DuneSQL. We will not accept contributions to the lineage of the following models:

* dex.trades
* nft.trades
* labels
* token.erc20
* tokens.nft

Run the following command to see the list of affected files:

```
dbt ls --resource-type model --output path --select +dex_trades +labels +nft_trades +tokens_nft +tokens_erc20
```

Don't hesitate to reach out on Discord if you have any questions.

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by danielpartida** [comment]
@jeff-dude btw, I noticed that all the columns "address" for the schemas safe_celo.*, e.g. safe_celo.safes have the type varchar instead of varbinary. Did I miss something on the previous PR to specify the type?  I followed the other safe spells implementations. The schemas with this error are safe_celo.singletons, safe_celo.transactions, safe_celo.safes and safe_celo.transfers

Screenshots attached for safe_ethereum.safes and safe_celo.safes

Screenshot 2023-06-29 at 18 52 05

Screenshot 2023-06-29 at 18 51 36

**Comment by jeff-dude** [comment]
> @jeff-dude btw, I noticed that all the columns "address" and "tx_hash" for the schemas safe_celo.*, e.g. safe_celo.safes have the type varchar instead of varbinary. Did I miss something on the previous PR to specify the type? I followed the other safe spells implementations. The schemas with this error are safe_celo.singletons, safe_celo.transactions, safe_celo.safes and safe_celo.transfers
> 
> Screenshots attached for safe_ethereum.safes and safe_celo.safes
> 
> Screenshot 2023-06-29 at 18 52 05 Screenshot 2023-06-29 at 18 51 36

since these are still built on the spark engine, they don't have the built-in datatypes such as `varbinary`, enabled on dunesql. in the past, we've used a manual override process on the query execution side to cast data types on the fly. as we migrate spellbook to dunesql the next few days, the data types will be available out-of-box (and inferred by source tables, which all base/decoded should have varbinary as expected).

is this something that can wait until early next week, to build on dunesql engine?

**Comment by danielpartida** [comment]
Yes, of course, it was more of a question of whether I needed to fix something :) Thank you for the explanation!

**Comment by danielpartida** [comment]
Let me know if you need additional input from my end @jeff-dude

File: models/safe/celo/safe_celo_schema.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    description: "Singletons addresses used with Safes"
+    columns:
+      - &address
+        name: address
+        description: "Safe contract address"
+        tests:
+          - unique
+          - not_null
+
+  - name: safe_celo_safes
```
Comment (jeff-dude) [inline models/safe/celo/safe_celo_schema.yml:23]: can we add the unique combo of columns test?

File: models/safe/celo/safe_celo_schema.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        description: "Initial version during creation"
+      - &block_date
+        name: block_date
+      - &creation_time
+        name: creation_time
+        description: "Datetime of Safe creation"
+      - &tx_hash
+        name: tx_hash
+
+  - name: safe_celo_transactions
```
Comment (jeff-dude) [inline models/safe/celo/safe_celo_schema.yml:50]: can we add the unique combo of columns test?

**Review by jeff-dude — APPROVED** [review]
two minor comments below.

you can also look to start building this on dunesql 🙌

File: models/safe/celo/safe_celo_schema.yml (hunk 1 above)
Comment (danielpartida) [reply models/safe/celo/safe_celo_schema.yml:23]: Hi @jeff-dude, thank you for checking this PR. Two questions

1. What should I pay attention to when building this on dunesql?
2. Not sure what you mean by "unique combo of columns test".

Comment (jeff-dude) [reply models/safe/celo/safe_celo_schema.yml]: @danielpartida following up here if you plan to proceed on this PR?

Comment (danielpartida) [reply models/safe/celo/safe_celo_schema.yml]: Yes, 100%. 

tbh, but I have not fully understood the meaning behind the unique combo test. I was looking for documentation on the migration for DuneSQL spells but did not find something useful yet. I just have not had the time to look at the  DuneSQL Migration tag PR to have an idea of what to change.

Comment (jeff-dude) [reply models/safe/celo/safe_celo_schema.yml]: okay, let me help on this PR and we'll keep the scope narrow to adding celo on legacy spark.

as for the migration, please check out the readme, we put migration info at the top. hopefully that gives a good review.

**Review by jeff-dude — APPROVED** [review]
okay, we're good here, notice i moved to the legacy files for spark. a follow up PR, you can use the non-legacy files to tranlsate to dunesql

**Comment by danielpartida** [comment]
Thank you ser

**Pull request merged**